pkg/controller/node/controller_utils.go

```diff
@@ -110,6 +111,10 @@ func deletePods(kubeClient clientset.Interface, recorder record.EventRecorder, n
 		if err == nil { // No error means at least one daemonset was found
 			continue
 		}
+		// if the pod is forgivable (wants to stay bound on the node), ignore it
+		if isPodForgivable(pod, nodeStatus.readyTransitionTimestamp) {
```

Comment:
Delete pods is called at many points - not just at that point.  Also, you are in a controller and cannot safely rely on the timestamp to guarantee the forgiveness period has been met (because of the potential for clock skew with the masters).  The node controller really needs to maintain its own clock and measure forgiveness from the time it observes a down node.